include/llbuild/Basic/FileInfo.h

```diff
@@ -75,6 +98,8 @@ struct FileInfo {
   uint64_t size;
   /// The modification time of the file.
   FileTimestamp modTime;
+  /// The MD5 hash of the file.
+  LazyMD5Digest digest;
```

Comment:
Similar to @ddunbar's comment, I think we'll want this to be selectable by the client.  Probably from a layering perspective, we should have a separate digest build value that holds an identifier and a digest value (blob?).  The lazy calculation is a good idea, but likely should be done outside of FileInfo, using a task in the build graph.   That will separate the concerns better and allow the system to be more explicit in when it is calculated.  Right now there is no way for the system to be sure that we don't accidentally calculate the digest of the file at a much later time when the rest of the info has possibly changed.

Given the last point, I think the digest task should probably be formulated such that it outputs two build values, the FileInfo of the file at digest calculation and the Digest value itself.  i.e. `fstat()` the same `fd` we use.  That will make sure we guard against outside changes impacting our view of the external (file system) environment.